src/ucs/sys/sys.c

```diff
@@ -917,20 +917,19 @@ void ucs_sys_free(void *ptr, size_t length)
     }
 }
 
-const char* ucs_sys_get_mem_lock_limit(char *str, size_t max_size)
+char* ucs_sys_get_mem_lock_limit(char *str, size_t max_size)
 {
     struct rlimit limit_info;
 
     /* Return the value of the max locked memory which is set on the system
      * (ulimit -l) */
     if (getrlimit(RLIMIT_MEMLOCK, &limit_info) != 0) {
-        ucs_error("getrlimit() failed: %m. Unable to read system limits");
-        return "Invalid String";
+        return "<unknown>";
     } else {
         if (limit_info.rlim_cur == RLIM_INFINITY) {
             ucs_snprintf_zero(str, max_size, "%s", "'unlimited'");
         } else {
-            ucs_snprintf_zero(str, max_size, "%lu", limit_info.rlim_cur);
+            ucs_snprintf_zero(str, max_size, "%llu", limit_info.rlim_cur / UCS_KBYTE);
```

Comment:
can use ucs_memunits_to_str()

Reply:
@yosefe IMHO we should print the value as it is shown by 'ulimit -l' : 
$ulimit -l
16384

Reply:
then we should add k/kB suffix for clarity

Reply:
ok, will add kbytes (like ulimit states)
`max locked memory       (kbytes, -l) 16384`

`ib_md.c:334  UCX  ERROR ibv_reg_mr(address=0x7f8a70a00000, length=37748736, access=0xf) failed: Cannot allocate memory. Please set max locked memory (ulimit -l) to 'unlimited' (current: 16384 kbytes)
`